src/pal/pal_fix_calling_convention_pass.h

```diff
@@ -0,0 +1,14 @@
+// Copyright (c) 2023 Advanced Micro Devices, Inc. All rights reserved.
+
+#ifndef PAL_PLATFORM_FIX_CALLING_CONVENTION_H
+#define PAL_PLATFORM_FIX_CALLING_CONVENTION_H
+
+#include <llvm/IR/PassManager.h>
+
+/// This pass sets the calling convention to AMDGPU_Gfx for all calls in the given module and sets the AlwaysInline
+/// Attribute on every called function in the module to avoid the LLVM AMDGPU backend throwing errors.
```

Comment:
This is because PAL doesn't support a call stack, right ? But can PAL support (stackless) continuations, ie one-way jumping to another function ?

Reply:
This is not a PAL thing, but rather a LLVM AMDGPU backend thing. Jumps are supported in theory but we ran into issues with the ocml and other library functions. To be on the safe side we force-inline everything in the IR and then the backend is happy. Since our focus was to get things up and running we didn't spend more time on this problem yet.